### ISSUE TITLE
Virtualization mode in Center and XOA differ

### ISSUE THREAD

**Issue by Velociraptor85** [xcp-ng/xenadmin#148]
**Describe the bug**
XOA 5.33.1 shows PVHVM as the virtualisation mode for some specific virtual guest but for the same guest XCP-ng Center 7.6.3 shows HVM. So I am unsure if that is because Center does not know the difference or maybe the box does not report it correctly but it is consistent for other virtualized guests so it does not seem to be a machine specific issue. 

I am expecting this to show the same information.

**To Reproduce**
XOA:
Home->VMs -> "anyVM" -> Advanced -> Virtualisation mode:
Center:
"anyVM" -> General -> Virtualisation mode: 

**Screenshots / Logfiles**
XOA:
![grafik](https://user-images.githubusercontent.com/4143960/55886711-8d061780-5bac-11e9-9862-4a54678d24ce.png)

Center: 
![grafik](https://user-images.githubusercontent.com/4143960/55886631-6a73fe80-5bac-11e9-99a0-772c7c627fa9.png)

**Comment by olivierlambert** [comment]
That's because (I think) XCP-ng Center is not aware of the PV drivers status (installed or not)

**Comment by Velociraptor85** [comment]
Well is this than classified as a bug or a feature request :?

**Comment by olivierlambert** [comment]
I don't see the reason to call this a bug. You are basically asking for adding PV drivers detection in XCP-ng to display the "detailed" mode.

**Comment by Velociraptor85** [comment]
Well for me it seems strange that a variable "Virtualization mode" has differen values although its the same object :? But for me if this is not concidered a bug we can close this issue.

**Comment by olivierlambert** [comment]
That's because of a misconception of what you imagine about those XAPI clients.

For example, in XO displaying virtualization mode: we don't display the "raw" field from XAPI. In real Xen world, there is only 2 virt mode: PV or HVM. However, people often asked to get more details than this. So we combined info from 2 different fields: virt mode and PV driver status. The logic is simple: if you detect HVM mode + PV drivers = PVHVM.

Long story short: there is no XAPI client that just displays XAPI raw values, because they are often very low level and not really useful.

**Comment by Velociraptor85** [comment]
Wouldnt it be maybe a better idea to split it into two? Like "Virtualization Mode": HVM/PV and than a second Variable which states that the PVDrivers are installed ?

**Comment by olivierlambert** [comment]
Why that? PVHVM is used widely to say exactly that: it's a HVM guest with PV drivers inside.

And this is exactly the text displayed in XO:

> Hardware virtualization with paravirtualization drivers enabled (PVHVM)

**Comment by Velociraptor85** [comment]
Good ^^ was just an idea ;) thanks for clarifyiing

**Comment by schulz-alexander** [comment]
I will not work on that until I have more free time :hourglass_flowing_sand: 
If someone else has time, feel free to contribute :-)

Hint: I assume there is some weird code to explore, because the fields are created in contraint to the template the VM was created from.

**Comment by schulz-alexander** [comment]
we will not work on that
